src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,30 +36,66 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        addressFields = trimmedAddress.split(FIELD_SEPARATOR);
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(addressFields)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        
+        setFields(addressFields);
     }
 
     /**
      * Returns true if a given string is a valid person email.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public static boolean isValidAddress(String[] test) {
+        return test.length == TOTAL_FIELDS && PostalCode.isValidPostalCode(test[POSTAL_CODE_INDEX]);
     }
 
+    private void setFields(String[] fields) {
+    	try {
+			this.block = new Block(fields[BLOCK_INDEX]);
+			this.street = new Street(fields[STREET_INDEX]);
+	    	this.unit = new Unit(fields[UNIT_INDEX]);
+	    	this.postalCode = new PostalCode(fields[POSTAL_CODE_INDEX]);
+		} catch (IllegalValueException e) {
+			e.printStackTrace();
```

Comment:
this is hardly any exception handling! 
catch block should be doing something meaningful  rather than print stack trace :-P